file_catalog/mongo.py

```diff
@@ -40,7 +40,6 @@ def __init__(self, host=None, port=None, authSource=None, username=None, passwor
         self.client.files.create_index([('logical_name',pymongo.HASHED)], background=True)
         self.client.files.create_index('locations', unique=True, background=True)
         self.client.files.create_index([('locations.site',pymongo.DESCENDING),('locations.path',pymongo.DESCENDING)], background=True)
-        self.client.files.create_index('locations.archive', background=True)
```

Comment:
do you think the problem was that locations was double-indexed? One on archive and the other on site/path?

Reply:
Possibly, but the "correct" index for the query in question was on `logical_name`.

Reply:
It could even be that we're running such an old version of Mongo (3.0), that the query planner wasn't smart enough, and just picked the first index it found.

Reply:
I see. Picking the "right" index would be difficult without prior knowledge I assume

Reply:
Is there no way of defining an order of precedence for indices?

Reply:
Nope, that's not something DBs do.  The best you can do is to "hint" at an index to use when you call a query.  

The DB's query analyzer is supposed to be smart enough to pick the "most optimal" index, but Mongo 3.0 was still relatively young for a DB, unlike MySQL.